pennylane/devices/default_clifford.py

```diff
@@ -460,57 +519,239 @@ def simulate(
                         flat_state = qml.math.flatten(state)
                         debugger.snapshots[op.tag or len(debugger.snapshots)] = flat_state
 
+        stim_circuit = initial_circuit + stim_ct
         tableau_simulator.do_circuit(stim_ct)
-
         global_phase = qml.GlobalPhase(qml.math.sum(op.data[0] for op in global_phase_ops))
-        return self.measure(circuit, tableau_simulator, global_phase, stim)
+
+        # Perform measurments based on whether shots are provided
+        if circuit.shots:
+            meas_results = self.measure_statistical(circuit, stim_circuit, stim, seed=seed)
+        else:
+            meas_results = self.measure_analytical(
+                circuit, stim_circuit, tableau_simulator, global_phase, stim
+            )
+
+        return meas_results[0] if len(meas_results) == 1 else tuple(meas_results)
 
     @staticmethod
-    def pl_to_stim(op):
+    def _pl_to_stim(op):
         """Convert PennyLane operation to a Stim operation"""
         try:
             stim_op = _GATE_OPERATIONS[op.name]
         except KeyError as e:
-            raise DeviceError(
+            raise qml.DeviceError(
                 f"Operator {op} not supported on default.clifford and does not provide a decomposition."
             ) from e
         return stim_op, " ".join(map(str, op.wires))
 
-    def measure(self, circuit, tableau_simulator, global_phase, stim):
-        """Given a circuit, compute and return the measurement results."""
+    # pylint:disable=too-many-return-statements
+    def _convert_op_to_linear_comb(self, meas_obs, coeffs, paulis):
+        """Convert a PennyLane observable to a linear combination of stim Pauli terms"""
+
+        # Case for simple Pauli terms
+        if isinstance(meas_obs, (qml.Identity, qml.PauliZ, qml.PauliX, qml.PauliY)):
+            coeffs.append(1.0)
+            paulis.append((_GATE_OPERATIONS[meas_obs.name], meas_obs.wires))
+            return coeffs, paulis
+
+        # Case for Sum
+        if isinstance(meas_obs, Sum):
+            for op in meas_obs:
+                coeffs, paulis = self._convert_op_to_linear_comb(op, coeffs, paulis)
+            return coeffs, paulis
+
+        # Case for higher arithmetic depth for prod-type observables
+        if meas_obs.arithmetic_depth > 1 and isinstance(meas_obs, (Prod, SProd)):
+            with qml.operation.use_new_opmath():
+                meas_obs_simp = meas_obs.simplify()
```

Comment:
Why the need to call simplify inside the `use_new_opmath()` context?  This sounds like a bug we might need to fix.

Reply:
I don't think we need to anymore, this code pre-dates the fixes I pushed

Reply:
Removed this!